olive/cli/quantize.py

```diff
@@ -0,0 +1,132 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+# ruff: noqa: T201
+# ruff: noqa: RUF012
+
+import json
+from argparse import ArgumentParser
+from copy import deepcopy
+from typing import Any, Dict
+
+from olive.cli.base import BaseOliveCLICommand
+from olive.common.utils import set_nested_dict_value
+from olive.workflows import run as olive_run
+
+
+class QuantizeCommand(BaseOliveCLICommand):
+    _CONFIG_TEMPLATE = {
+        "input_model": {"type": "HfModel", "load_kwargs": {"attn_implementation": "eager"}},
+        "systems": {
+            "local_system": {
+                "type": "LocalSystem",
+                "accelerators": [{"device": "gpu", "execution_providers": ["CUDAExecutionProvider"]}],
+            }
+        },
+        "data_configs": [],
+        "passes": {
+            "awq": {"type": "AutoAWQQuantizer"},
+            "gptq": {
+                # Ref: https://github.com/AutoGPTQ/AutoGPTQ/pull/651/files
+                "type": "GptqQuantizer",
+                "data_config": None,
+                "outside_layer_modules": ["model.embed_tokens", "embed_dropout", "model.norm"],
```

Comment:
Maybe we should not hardcode these values which are for a specific hf model type? We could instead use the mappings in https://github.com/microsoft/Olive/blob/main/olive/common/hf/mappings.py to add support for phi3 in the pass.